diff --git a/bazel/upb_c_proto_library.bzl b/bazel/upb_c_proto_library.bzl
--- a/bazel/upb_c_proto_library.bzl
+++ b/bazel/upb_c_proto_library.bzl
@@ -1,3 +1,4 @@
 """upb_c_proto_library() exposes upb's generated C API for protobuf (foo.upb.h)"""
 
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 load("//bazel:upb_minitable_proto_library.bzl", "UpbMinitableCcInfo", "upb_minitable_proto_library_aspect")
diff --git a/bazel/upb_minitable_proto_library.bzl b/bazel/upb_minitable_proto_library.bzl
--- a/bazel/upb_minitable_proto_library.bzl
+++ b/bazel/upb_minitable_proto_library.bzl
@@ -1,3 +1,4 @@
 """upb_minitable_proto_library() exposes upb's generated minitables (foo.upb_minitable.h)"""
 
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 load("//bazel/common:proto_info.bzl", "ProtoInfo")
diff --git a/bazel/upb_proto_reflection_library.bzl b/bazel/upb_proto_reflection_library.bzl
--- a/bazel/upb_proto_reflection_library.bzl
+++ b/bazel/upb_proto_reflection_library.bzl
@@ -1,3 +1,4 @@
 """upb_c_proto_reflection_library() exposes upb reflection for protobuf (foo.upbdefs.h)"""
 
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 load("//bazel:upb_minitable_proto_library.bzl", "UpbMinitableCcInfo", "upb_minitable_proto_library_aspect")
diff --git a/bazel/upb_proto_library.bzl b/bazel/upb_proto_library.bzl
--- a/bazel/upb_proto_library.bzl
+++ b/bazel/upb_proto_library.bzl
@@ -1,3 +1,4 @@
 # Copyright (c) 2009-2021, Google LLC
 # All rights reserved.
 #
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
diff --git a/bazel/private/upb_proto_library_internal/cc_library_func.bzl b/bazel/private/upb_proto_library_internal/cc_library_func.bzl
--- a/bazel/private/upb_proto_library_internal/cc_library_func.bzl
+++ b/bazel/private/upb_proto_library_internal/cc_library_func.bzl
@@ -1,4 +1,5 @@
 """A function to compile C/C++ code, like cc_library() but from Starlark."""
 
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain", "use_cpp_toolchain")
 
diff --git a/bazel/private/upb_proto_library_internal/aspect.bzl b/bazel/private/upb_proto_library_internal/aspect.bzl
--- a/bazel/private/upb_proto_library_internal/aspect.bzl
+++ b/bazel/private/upb_proto_library_internal/aspect.bzl
@@ -1,3 +1,4 @@
 """Implementation of the aspect that powers the upb_*_proto_library() rules."""
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 
 load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")
diff --git a/bazel/private/upb_proto_library_internal/rule.bzl b/bazel/private/upb_proto_library_internal/rule.bzl
--- a/bazel/private/upb_proto_library_internal/rule.bzl
+++ b/bazel/private/upb_proto_library_internal/rule.bzl
@@ -1,3 +1,4 @@
 """Internal rule implementation for upb_*_proto_library() rules."""
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 
 def _filter_none(elems):
diff --git a/build_defs/cc_proto_blacklist_test.bzl b/build_defs/cc_proto_blacklist_test.bzl
--- a/build_defs/cc_proto_blacklist_test.bzl
+++ b/build_defs/cc_proto_blacklist_test.bzl
@@ -1,3 +1,4 @@
 """Contains a unittest to verify that `cc_proto_library` does not generate code for blacklisted `.proto` sources (i.e. WKPs)."""
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 
 load("@bazel_skylib//lib:unittest.bzl", "asserts", "unittest")
diff --git a/hpb/bazel/hpb_proto_library.bzl b/hpb/bazel/hpb_proto_library.bzl
--- a/hpb/bazel/hpb_proto_library.bzl
+++ b/hpb/bazel/hpb_proto_library.bzl
@@ -1,3 +1,4 @@
 # Copyright (c) 2024 Google LLC
 # All rights reserved.
 #
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
diff --git a/pkg/build_systems.bzl b/pkg/build_systems.bzl
--- a/pkg/build_systems.bzl
+++ b/pkg/build_systems.bzl
@@ -1,3 +1,4 @@
 """Starlark utilities for working with other build systems."""
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 
 load("@bazel_skylib//lib:paths.bzl", "paths")
diff --git a/pkg/cc_dist_library.bzl b/pkg/cc_dist_library.bzl
--- a/pkg/cc_dist_library.bzl
+++ b/pkg/cc_dist_library.bzl
@@ -1,3 +1,4 @@
 # Rules for distributable C++ libraries
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 
 load("@rules_cc//cc:action_names.bzl", cc_action_names = "ACTION_NAMES")
diff --git a/rust/bazel/aspects.bzl b/rust/bazel/aspects.bzl
--- a/rust/bazel/aspects.bzl
+++ b/rust/bazel/aspects.bzl
@@ -1,3 +1,4 @@
 """This file implements rust_proto_library aspect."""
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 
 load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")
diff --git a/rust/test/rust_proto_library_unit_test/rust_proto_library_unit_test.bzl b/rust/test/rust_proto_library_unit_test/rust_proto_library_unit_test.bzl
--- a/rust/test/rust_proto_library_unit_test/rust_proto_library_unit_test.bzl
+++ b/rust/test/rust_proto_library_unit_test/rust_proto_library_unit_test.bzl
@@ -1,3 +1,4 @@
 """This module contains unit tests for rust_proto_library and its aspect."""
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 
 load("@bazel_skylib//lib:unittest.bzl", "analysistest", "asserts")
